### PR TITLE
all geometry entities have evalf capability

### LINKED ISSUE #22109: evalf(), .n or N of Point2D produces error.
The following code produces the error `TypeError: evalf() got multiple values for argument 'n'`.  I would expect it to produce `𝑃𝑜𝑖𝑛𝑡2𝐷(6,6)`

```python
from sympy import N, symbols
from sympy.geometry import Point2D, Segment2D

a,b,c = symbols('a,b,c')

eq1 = a + b + c
subs = {a:1,b:2,c:3}
eq1.n(n=5, subs=subs)

eq2 = a * b * c
eq2.n(n=5, subs=subs)

p = Point2D(eq1 ,eq2)
p.n(n=5, subs=subs)
```

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

closes #22110 as a continuation
fixes #22109

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by smichr** [comment]
@oscarbenjamin , does this look complete, now?

File: sympy/geometry/ellipse.py [hunk 1]
```diff
@@ -1605,6 +1606,13 @@ def __new__(cls, *args, **kwargs):
 
             raise GeometryError("Circle.__new__ received unknown arguments")
 
+    def _eval_evalf(self, prec=15, **options):
+        pt, r = self.args
+        dps = prec_to_dps(prec)
+        pt = pt.evalf(n=dps, **options)
+        r = r.evalf(n=dps, **options)
+        return self.func(pt, r, evaluate=False)
```
Comment (praneethratna) [inline sympy/geometry/ellipse.py:1614]: @smichr Can this be written something like `self.func(*[x.evalf(n=prec_to_dps(prec), **options) for x in self.args], evaluate = False)` similar to one in Point class ?

Comment (oscarbenjamin) [reply sympy/geometry/ellipse.py:1614]: This method computes `dps` only once which is good. Ideally `_eval_evalf` would be able to call a method that takes binary precision. There is a PR for that but it is unfinished  #20493

**Comment by oscarbenjamin** [comment]
Looks good to me

**Pull request merged**